### PR TITLE
Fix Wikimedia Commons image fetch 502 by setting User-Agent

### LINKED ISSUE #340: Add Wikidata as image source via Wikimedia Commons
`[mode: plan]` `[model: sonnet]` [effort: medium]

## Problem

The Wikidata provider (`internal/provider/wikidata/`) currently returns `nil, nil` from `GetImages()`. However, Wikidata items frequently link to Wikimedia Commons images via:

- **P18** (image) -- typically a portrait/photo of the artist
- **P154** (logo image) -- the artist/band logo

These images are freely licensed and available at various resolutions through the Wikimedia Commons API.

## Proposed Solution

Extend the Wikidata provider to fetch image properties from SPARQL and resolve them to downloadable Wikimedia Commons URLs.

### 1. Extend the SPARQL Query

```sparql
SELECT ?item ?itemLabel ?inception ?dissolved ?countryLabel ?genreLabel ?image ?logo WHERE {
  ?item wdt:P434 "{mbid}" .
  OPTIONAL { ?item wdt:P571 ?inception . }
  OPTIONAL { ?item wdt:P576 ?dissolved . }
  OPTIONAL { ?item wdt:P495 ?country . }
  OPTIONAL { ?item wdt:P136 ?genre . }
  OPTIONAL { ?item wdt:P18 ?image . }
  OPTIONAL { ?item wdt:P154 ?logo . }
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en" . }
}
```

### 2. Resolve Wikimedia Commons URLs

Wikimedia Commons filenames (e.g., `File:Radiohead.jpg`) need to be resolved to actual image URLs. The Commons API provides this:

```
https://commons.wikimedia.org/w/api.php?action=query&titles=File:{filename}&prop=imageinfo&iiprop=url|size&format=json
```

This returns the full-resolution URL and dimensions.

For thumbnails at a specific width:
```
https://commons.wikimedia.org/w/api.php?action=query&titles=File:{filename}&prop=imageinfo&iiprop=url|size&iiurlwidth=1000&format=json
```

### 3. Implementation Changes

**`internal/provider/wikidata/wikidata.go`:**

```go
// GetImages fetches artist images from Wikimedia Commons via Wikidata properties.
func (a *Adapter) GetImages(ctx context.Context, mbid string) ([]provider.ImageResult, error) {
    if err := a.limiter.Wait(ctx, provider.NameWikidata); err != nil {
        return nil, &provider.ErrProviderUnavailable{
            Provider: provider.NameWikidata,
            Cause:    fmt.Errorf("rate limiter: %w", err),
        }
    }

    query := buildImageQuery(mbid)
    bindings, err := a.executeSPARQL(ctx, query)
    if err != nil {
        return nil, err
    }

    var results []provider.ImageResult
    for _, b := range bindings {
        if b.Image.Value != "" {
            url, w, h, err := a.resolveCommonsURL(ctx, extractFilename(b.Image.Value))
            if err == nil {
                results = append(results, provider.ImageResult{
                    URL:    url,
                    Type:   provider.ImageThumb,
                    Width:  w,
                    Height: h,
                    Source: string(provider.NameWikidata),
                })
            }
        }
        if b.Logo.Value != "" {
            url, w, h, err := a.resolveCommonsURL(ctx, extractFilename(b.Logo.Value))
            if err == nil {
                results = append(results, provider.ImageResult{
                    URL:    url,
                    Type:   provider.ImageLogo,
                    Width:  w,
                    Height: h,
                    Source: string(provider.NameWikidata),
                })
            }
        }
    }
    return results, nil
}
```

**`internal/provider/wikidata/types.go`:**
- Add `Image` and `Logo` fields to `SPARQLBinding`

### 4. Update Field Priorities

In `internal/provider/settings.go`, add Wikidata to image field defaults:

```go
// Current: thumb: Fanart.tv, AudioDB, Deezer, Spotify
// New:     thumb: Fanart.tv, AudioDB, Deezer, Spotify, Wikidata
// Current: logo:  Fanart.tv, AudioDB
// New:     logo:  Fanart.tv, AudioDB, Wikidata
```

Wikidata should be last in priority (lower resolution than dedicated image providers).

### 5. Rate Limiting

The Wikimedia Commons API shares the same rate limit as Wikidata SPARQL. A single `GetImages` call may require 1 SPARQL query + up to 2 Commons API calls. Ensure rate limiter accounts for this (current 5 req/sec should be sufficient with burst).

## Testing

- Unit tests with mocked SPARQL responses containing P18/P154 values
- Unit tests with mocked Commons API responses
- Test edge cases: no image property, invalid filename, Commons API failure
- Integration test: verify image URLs are valid and downloadable for a known artist (e.g., Q44190 / Radiohead)
- Docker UAT: verify Wikidata images appear in the image selection UI

## Acceptance Criteria

- [ ] `GetImages()` returns thumb images from P18 (image property)
- [ ] `GetImages()` returns logo images from P154 (logo image property)
- [ ] Wikimedia Commons URLs are properly resolved with dimensions
- [ ] Field priorities updated to include Wikidata for thumb and logo
- [ ] Rate limiting covers both SPARQL and Commons API calls
- [ ] Graceful handling of missing images, Commons API errors
- [ ] Tests cover SPARQL response parsing and Commons URL resolution

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

- Add `User-Agent: Stillwater/1.0` header to `fetchImageFromURL` and `ProbeRemoteImage`, which were using Go's default `Go-http-client/1.1` -- blocked by Wikimedia Commons
- The Wikidata provider already set the header for SPARQL and Commons API calls, but the downstream image download paths did not propagate it
- Add regression test `TestProbeRemoteImage_UserAgent` to verify the header is sent

Closes #340

## Test plan

- [ ] Run `go test ./internal/api/ ./internal/image/` -- all pass including new User-Agent test
- [ ] Deploy and fetch a Wikidata image (e.g. thumb for an artist with a Wikidata image) -- should return 200 instead of 502

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved external image fetching to use a more robust HTTP request flow and set a fixed User-Agent to avoid remote blocks.

* **Tests**
  * Added test coverage verifying the outbound User-Agent is sent when probing remote images.

### CODE REVIEW

File: internal/api/handlers_image.go [hunk 1]
```diff
@@ -664,7 +664,14 @@ func ssrfSafeTransport() *http.Transport {
...
 	client := r.ssrfClient
 
-	resp, err := client.Get(rawURL) //nolint:gosec,noctx // G107: URL is validated by caller; background fetch is acceptable
+	req, err := http.NewRequest(http.MethodGet, rawURL, nil) //nolint:gosec,noctx // G107/G704: URL is validated by caller; background fetch is acceptable
+	if err != nil {
+		return nil, fmt.Errorf("creating request: %w", err)
+	}
+	// Wikimedia Commons blocks requests without a proper User-Agent.
+	req.Header.Set("User-Agent", "Stillwater/1.0 (https://github.com/sydlexius/stillwater)")
+
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_image.go]: Established pattern in the codebase -- six provider packages each define their own User-Agent string (some use the full URL, some just "Stillwater/1.0", MusicBrainz uses version.Version dynamically, AllMusic/DuckDuckGo use browser UAs). Centralizing would be a broader refactor across all providers, not suited for a bug-fix PR.

File: internal/image/processor.go [hunk 2]
```diff
@@ -34,6 +34,8 @@ func ProbeRemoteImage(ctx context.Context, rawURL string) (*RemoteImageInfo, err
 	if err != nil {
 		return nil, fmt.Errorf("creating request: %w", err)
 	}
+	// Wikimedia Commons blocks requests without a proper User-Agent.
+	req.Header.Set("User-Agent", "Stillwater/1.0 (https://github.com/sydlexius/stillwater)")
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/image/processor.go]: Same as above -- consistent with the existing per-package User-Agent pattern. Centralizing across all providers is out of scope for this fix.

File: internal/image/processor_test.go [hunk 3]
```diff
@@ -361,6 +361,27 @@ func TestProbeRemoteImage(t *testing.T) {
 	}
 }
 
+func TestProbeRemoteImage_UserAgent(t *testing.T) {
+	data := makeJPEG(t, 64, 64)
+	var gotUA string
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotUA = r.Header.Get("User-Agent")
+		w.Header().Set("Content-Type", "image/jpeg")
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/image/processor_test.go]: Fixed in 4d61a4a. Replaced bare variable with a buffered channel to synchronize the handler goroutine write with the test goroutine read.
